logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/MessageRepository.kt

```diff
@@ -151,6 +149,17 @@ interface MessageRepository {
         conversationId: ConversationId
     ): Either<CoreFailure, Conversation.ReceiptMode?>
 
+    /**
+     * updates the message status to [MessageEntity.Status.SENT] and the server date to [serverDate]
+     * also mark other pending messages and add millis to their date
+     */
+    suspend fun updateMessagesAfterOneIsSent(
```

Comment:
The naming here is tough...

What about `promoteMessageToSentUpdatingServerTime`.

Same thing for the query name.